crates/editor/src/signature_help_popover.rs

```diff
@@ -0,0 +1,553 @@
+
+    let signature_information = maybe_active_signature
+        .and_then(|active_signature| signature_information.get(active_signature as usize))
+        .or_else(|| signature_information.first())?;
+
+    let str_for_join = ", ";
+    let parameter_length = signature_information
+        .parameters
+        .as_ref()
+        .map(|parameters| parameters.len())
+        .unwrap_or(0);
+    let mut highlight_start = 0;
+    let (markdown, mut highlights): (Vec<_>, Vec<_>) = signature_information
+        .parameters
+        .as_ref()?
+        .iter()
+        .enumerate()
+        .filter_map(|(i, parameter_information)| {
+            let string = match parameter_information.label.clone() {
+                lsp::ParameterLabel::Simple(string) => string,
+                lsp::ParameterLabel::LabelOffsets(offset) => signature_information
+                    .label
+                    .chars()
+                    .skip(offset[0] as usize)
+                    .take((offset[1] - offset[0]) as usize)
+                    .collect::<String>(),
+            };
+            let string_length = string.len();
+
+            let result = if let Some(active_parameter) = maybe_active_parameter {
+                if i == active_parameter as usize {
+                    Some((
+                        string,
+                        Some((
+                            highlight_start..(highlight_start + string_length),
+                            SIGNATURE_HELP_HIGHLIGHT_CURRENT,
+                        )),
+                    ))
+                } else {
+                    Some((string, None))
+                }
+            } else {
+                Some((string, None))
+            };
+
+            if i != parameter_length {
+                highlight_start += string_length + str_for_join.len();
+            }
+
+            result
+        })
+        .unzip();
+    let markdown = markdown.join(str_for_join);
+
+    let language_name = language
+        .map(|n| n.name().to_lowercase())
+        .unwrap_or_default();
+
+    let markdown = if function_options_count >= 2 {
+        let suffix = format!("(+{} overload)", function_options_count - 1);
+        let highlight_start = markdown.len() + 1;
+        highlights.push(Some((
+            highlight_start..(highlight_start + suffix.len()),
+            SIGNATURE_HELP_HIGHLIGHT_OVERLOAD,
+        )));
+        format!("```{language_name}\n{markdown} {suffix}")
+    } else {
+        format!("```{language_name}\n{markdown}")
+    };
+
+    if markdown.is_empty() {
+        None
+    } else {
+        let highlights = highlights.into_iter().flatten().collect::<Vec<_>>();
+        Some((markdown, highlights))
+    }
+}
+
+impl SignatureHelpPopover {
+    pub fn render(
+        &mut self,
+        style: &EditorStyle,
+        max_size: Size<Pixels>,
+        workspace: Option<WeakView<Workspace>>,
+        cx: &mut ViewContext<Editor>,
+    ) -> AnyElement {
+        div()
```

Comment:
I do not think we __need__ to solve this edge case now, but it would definitely be nice (as all that we have to do is to limit the max horizontal shift for the panel?)

![image](https://github.com/zed-industries/zed/assets/2690773/1e78ff09-3541-407a-8802-6b13c05aa4ab)